climada/hazard/tc_tracks_synth.py

```diff
@@ -166,6 +206,170 @@ def _one_rnd_walk(track, ens_size, ens_amp0, ens_amp, max_angle, rnd_vec):
+    gamma = np.abs(np.mod(theta, np.pi) - \
+                   np.floor((np.mod(theta, np.pi) / (np.pi / 2)) + 0.5) * np.pi / 2)
+    # np.abs(dat['theta2'] - np.floor((dat['theta2'] / (np.pi / 2)) + 0.5) * np.pi / 2)
+    return 2 * np.sqrt(3) * (_f_ac(np.cos(theta) * x + np.sin(theta) * y, gamma) - 1 / 2)
+
+
+@jit
+def _f_ac(z, theta):
+    """F transform for autocorrelated random uniform series generation
+
+    This is function F defined here:
+    https://stats.stackexchange.com/questions/48086/algorithm-to-produce-autocorrelated-uniformly-distributed-number
+    The CDF of Y.
+
+    Parameters:
+        z (float): longitude coordinates, in decimal degrees
+        theta (float): arccos of the autocorrelation
+
+    Returns:
+        float (CDF at z)
+    """
+    c = np.cos(theta)
+    s = np.sin(theta)
+    if z >= np.sqrt(3) * (c + s):
+        res = 1
+    elif z > np.sqrt(3) * (c - s):
+        res = 1 / 12 / np.sin(2 * theta) * \
+              (-3 - z ** 2 + 2 * np.sqrt(3) * z * (c + s) + 9 * np.sin(2 * theta))
+    elif z > np.sqrt(3) * (-c + s):
+        res = 1 / 6 * (3 + np.sqrt(3) * z / c)
+    elif z > -np.sqrt(3) * (c + s):
+        res = 1 / 12 / np.sin(2 * theta) * \
+              (z ** 2 + 2 * np.sqrt(3) * z * (c + s) + 3 * (1 + np.sin(2 * theta)))
+    else:
+        res = 0
+    return res
+
+
+@jit
+def _get_bearing_angle(lon, lat):
+    """Compute bearing angle of great circle paths defined by consecutive points
+
+    Returns initial bearing (also called forward azimuth) of the n-1 great circle
+    paths define by n consecutive longitude/latitude points. The bearing is the angle
+    (clockwise from North) which if followed in a straight line along a great-circle
+    arc will take you from the start point to the end point. See also:
+    http://www.movable-type.co.uk/scripts/latlong.html
+    Here, the bearing of each pair of consecutive points is computed.
+
+    Parameters:
+        lon (numpy.ndarray of shape (n,)): longitude coordinates, in decimal degrees
+        lat (numpy.ndarray of shape (n,)): latitude coordinates, in decimal degrees
+
+    Returns:
+        bearing_angle (numpy.ndarray of shape (n-1,), in decimal degrees
+    """
+    lon, lat = map(np.radians, [lon, lat])
+    lat_1 = lat[:-1]
+    lon_1 = lon[:-1]
+    lat_2 = lat[1:]
+    lon_2 = lon[1:]
+    delta_lon = lon_2 - lon_1
+    # what to do with the points that don't move?
+    #   i.e. where lat_2=lat_1 and lon_2=lon_1? The angle does not matter in
+    # that case because angular distance will be 0.
+    earth_ang_fix = np.arctan2(np.sin(delta_lon) * np.cos(lat_2),
+                               np.cos(lat_1) * np.sin(lat_2) - \
+                               np.sin(lat_1) * np.cos(lat_2) * np.cos(delta_lon))
+    return np.degrees(earth_ang_fix)
+
+
+@jit
+def _get_angular_distance(lon, lat):
+    """Compute the angular distance of great circle paths defined by consecutive points
```

Comment:
There is a function `dist_approx` in `climada.util.coordinates` that computes approximate distances of points on earth. With the parameter `method="geosphere"`, it uses the Haversine formula. I suggest that you use that function instead of reimplementing the exact same functionality here.